colony.py

```diff
@@ -90,6 +90,9 @@ class Colony(object):
     def __init__(self):
         self._nodes = []
 
+    def get_nodes(self):
+        return self._nodes
```

Comment:
@jacksujingcheng I noticed this is only called to get the list of nodes (which can be done with `list(colony)`. But now that I think of it, it might better to implement an `__len__` magic function.
```suggestion
    def __len__(self):
        return len(self._nodes)
```